lib/AST/UnqualifiedLookup.cpp

```diff
@@ -487,6 +487,18 @@ void UnqualifiedLookupFactory::addImportedResults(const DeclContext *const dc) {
                  NLKind::UnqualifiedLookup, resolutionKind, dc,
                  Loc, nlOptions);
 
+  if (dc->isInSwiftinterface() &&
+      !dc->getASTContext().LangOpts.FormalCxxInteropMode) {
+    // It's possible that the textual interface was originally compiled without
```

Comment:
Could this cause any trouble with users who use swiftinterface files generated with an older version of the compiler (this we might wrongly assume it was compiler in C mode)? Is that a concern?

Reply:
It won't cause any more trouble than there already is.

If those swift interface files happen to be interop mode-agnostic, i.e., work with C++ interop either enabled or disabled, then they will still work as before.

If those interface files relied on C++ interop being disabled, then they will already broken, and hopefully this patch will fix them. 

If those interface files relied on C++ interop being enabled, this shouldn't affect them because, without the formal interop mode flag, the interop mode will simply follow that of the main compiler invocation (which needs to be C++ enabled).